### ISSUE TITLE
Suggestion: Trap `Interrupt` signal (Ctrl+C) and exit gracefully

### ISSUE THREAD

**Issue by DannyBen** [AlexB52/retest#124]
Right now, when pressing <kbd>Ctrl</kbd>+<kbd>C</kbd> after running a simple `retest`, exits with a ruby trace - sometimes a long one.

It would be nice to just see a friendly `Goodbye` or at least not see the trace.

```
$ retest
Setup identified: [RSPEC]. Using command: 'bundle exec rspec <test>'
Launching Retest...
^C
/home/vagrant/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rb-inotify-0.10.1
/lib/rb-inotify/notifier.rb:208:in `directory?': Interrupt
        from /home/vagrant/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rb-inotify-0.10.1/lib/rb-inotify/notifier.rb:208:in `block in watch'
        from /home/vagrant/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rb-inotify-0.10.1/lib/rb-inotify/notifier.rb:202:in `each'
        from /home/vagrant/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rb-inotify-0.10.1/lib/rb-inotify/notifier.rb:202:in `watch'

... snipped ...

```

(example rough fix in [this fork](https://github.com/DannyBen/retest/compare/main..dev))

**Comment by AlexB52** [comment]
Hi @DannyBen 
Thanks for the issue. This sounds like a really good idea. 
The change in your fork seems straight forward and I could add this feature to the next release :+1:

Do you want to create a pull request or are you ok with me pushing that forward? 

If you decide to create a pull request, please add a feature test in the `ruby-bare` app that simulates `ctrl`+`c`
Maybe using `Process.kill("INT", pid)` like in [Signal documentation](https://ruby-doc.org/core-2.5.0/Signal.html)

**Comment by DannyBen** [comment]
> Do you want to create a pull request or are you ok with me pushing that forward?

It will probably take you a fraction of the time it will take me to "round up" the implementation and tests, so feel free to go ahead and implement as and where you see fit.

**Comment by AlexB52** [comment]
> And BTW - you may have noticed I also removed and .gitignored the .ruby-version file.

Yeah I saw that too, thanks for pointing it out, it makes sense. I'll double check this separately at a later stage :+1:

**Comment by AlexB52** [comment]
Fixed in release 1.9.0
https://github.com/AlexB52/retest/releases

Thanks for these great suggestions @DannyBen

**Comment by DannyBen** [comment]
Thanks for implementing :)
